tools/check-upcoming-maintenance.py

```diff
@@ -0,0 +1,130 @@
+import os
+import re
+import subprocess
+from typing import List
+from optparse import OptionParser
+
+class NodeMaintenance:
+    '''
+    Class to keep track of project, zones, and search terms for nodes, as well as results from gcloud queries
+    '''
+    def __init__(self, project, zones, regex):
```

Comment:
I would provide default for `regex=None` to denote that it is optional.

Reply:
Done.